### PR TITLE
Added options to customise cpu and memory used per multicore for bismark align

### PR DESCRIPTION
**Pull request by ewels**
Closes nf-core/methylseq#124

Note that this PR will hopefully build with the new docker image created in PR #123. This should be considered when checking the CI tests..

### CODE REVIEW

**Comment by ewels** [comment]
Sigh, the docker build failed: https://cloud.docker.com/u/nfcore/repository/docker/nfcore/methylseq/builds/dce2a365-d736-4d56-a467-399b8e15ac36

```
UnsatisfiableError: The following specifications were found to be in conflict:
- bioconda::hisat2=2.1.0 -> python[version='>=2.7,<2.8.0a0']
- python=3.7
Use "conda search <package> --info" to see the dependencies for each package.
```

**Comment by ewels** [comment]
https://bioconda.github.io/contributor/guidelines.html states that:

> hisat2 runs `2to3` to make it Python 3 compatible, and copies over individual scripts to the bin dir

Sure enough, the build recipe does this: https://github.com/bioconda/bioconda-recipes/blob/90620569f00fa786b85d272507445a229748272d/recipes/hisat2/build.sh#L5-L20

However, the meta file then skips python 3k builds: https://github.com/bioconda/bioconda-recipes/blob/90620569f00fa786b85d272507445a229748272d/recipes/hisat2/meta.yaml#L13 🤦‍♂

**Comment by ewels** [comment]
x-ref: https://github.com/bioconda/bioconda-recipes/issues/18911

**Comment by ewels** [comment]
Reverted MultiQC back to version 1.7 and removed the Python 3 dependency. Hopefully everything will start working again then 🤞 

Still need to come back to the Python 3 thing at a later date, as future versions of MultiQC will drop py2k support. But we need to release soon to fix the iGenomes bug, so that can wait.

**Comment by ewels** [comment]
Again, needs to be merged and a new PR opened to test that the reverted MultiQC build works..

File: docs/usage.md [hunk 1]
```diff
@@ -347,6 +349,28 @@ Specify to run Bismark with the `--slam` flag to run bismark in [SLAM-seq mode](
 
 Specify to run Bismark with the `--local` flag to allow soft-clipping of reads. This should only be used with care in certain single-cell applications or PBAT libraries, which may produce chimeric read pairs. (See [Wu et al.](https://doi.org/10.1093/bioinformatics/btz125) (doesn't work with `--aligner bwameth`)
 
+### `--bismark_align_cpu_per_multicore`
+
+The pipeline makes use of the `--multicore` option for Bismark align. When using this option,
+Bismark uses a large number of CPUs for every `--multicore` specified. The pipeline
+calculates the number of `--multicore` based on the resources available to the task.
+It divides the available CPUs by 3, or by 5 if `--single_cell`, `--zymo` or `--non_directional`
```
Comment (alneberg) [inline docs/usage.md]: Does this mean 3 if either of the two first and 5 if `--non_directional` is specified? It's a bit ambiguous the way it's currently written

Comment (ewels) [reply docs/usage.md]: 3 unless any of those three are specified..

Comment (ewels) [reply docs/usage.md]: But yes I agree, this was not clear 👍 Hopefully improved now..

**Comment by ewels** [comment]
Tests look fine apart from MultiQC breaking..

File: docs/usage.md (hunk 1 above)
Comment (alneberg) [reply docs/usage.md]: Yes, much better.

**Pull request merged**